python/dask_cudf/dask_cudf/_legacy/io/parquet.py

```diff
@@ -348,8 +350,14 @@ def write_partition(
                 storage_options=kwargs.get("storage_options", None),
             )
         else:
-            with fs.open(fs.sep.join([path, filename]), mode="wb") as out_file:
-                if not isinstance(out_file, IOBase):
+            with (
+                contextlib.nullcontext()
+                if ioutils._is_local_filesystem(fs)
+                else fs.open(fs.sep.join([path, filename]), mode="wb")
+            ) as out_file:
+                if out_file is None:
+                    out_file = fs.sep.join([path, filename])
+                elif not isinstance(out_file, IOBase):
                     out_file = BufferedWriter(out_file)
```

Comment:
@ayushdg - I think we can remove this block? Do you remember why `BufferedWriter` was needed?